src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -247,20 +247,38 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
             resolver.resolve_str_path_error(DUMMY_SP, &path_str, ns, module_id)
         });
         debug!("{} resolved to {:?} in namespace {:?}", path_str, result, ns);
-        let result = match result {
-            Ok((_, Res::Err)) => Err(()),
-            x => x,
+        let result = match result.map(|(_, res)| res) {
+            Ok(Res::Err) | Err(()) => {
+                // resolver doesn't know about true and false so we'll have to resolve them
+                // manually as bool
+                if let Some((_, res)) = is_bool_value(path_str, ns) { Ok(res) } else { Err(()) }
```

Comment:
Can you revert this, in that case? We should be using `is_primitive` consistently.

Reply:
But is it really necessary to search through a list of strings for every `Res::Err`? Obviously I didn't measure it, but sounds like a waste. I'd like to ~sit on~ think about `is_primitive` uses for a short while before I revert it.

Reply:
Right now my biggest issue apart from performance is the WTF factor here. If I replace `is_bool_value` with `is_primitive` and leave the comment to document why it's necessary, I'm sure somebody will ask why we are looking through every primitive just to see if we have true or false right here.
I'm not looking for an excuse but I really feel that if we know that our search space has 2 elements or we have something fishy going on, we want the code to reflect that as well.

Reply:
So, I might have understood how rustc_resolve works. You're saying with the current code, it will properly resolve `bool`, `u64`, etc. even if we don't go through `is_primitive? If so having this check only for `is_bool_value` seems fine, but I'm not sure that's the case.

Reply:
> You're saying

I'm assuming - but I'd also be surprised if the call could resolve all sorts of complex paths, but not a primitive. I guess I need to verify my claim, though.

Reply:
An other, maybe even better solution would be to revert the `is_bool_value` related changes and run `is_primitive` before `rustc_resolve`. I see the same logic down in the function.